h/services/oauth.py

```diff
@@ -149,6 +149,25 @@ def create_token(self, user, authclient):
 
         return token
 
+    def create_grant_token(self, user, authclient):
+        """
+        Generate a JWT bearer grant token for a user.
+
+        :param user: The user to generate the token for.
+        :type user: h.models.User
+        :param authclient: The OAuth client that is going to use the token.
+        :type authclient: h.models.AuthClient
+        """
+        now = datetime.datetime.utcnow()
+        claims = {
+            'aud': self.domain,
+            'iss': authclient.id,
+            'sub': 'acct:{}@{}'.format(user.username, user.authority),
+            'nbf': now,
+            'exp': now + datetime.timedelta(minutes=5),
+        }
+        return jwt.encode(claims, authclient.secret, algorithm='HS256')
```

Comment:
Right. Note that the final implementation will most likely not use JWTs but instead a one-time authorisation code. I'm re-using them at the moment because it lets us get the front-end part working before the model changes & migrations are done.